src/Microsoft.AspNetCore.ResponseCaching/Internal/LoggerExtensions.cs

```diff
@@ -0,0 +1,322 @@
+                logLevel: LogLevel.Debug,
+                eventId: 16,
+                formatString: "Response is not cacheable since it contains the private cache directive.");
+            _logResponseWithUnsuccessfulStatusCodeNotCacheable = LoggerMessage.Define<int>(
+                logLevel: LogLevel.Debug,
+                eventId: 17,
+                formatString: "Response is not cacheable since its status code {StatusCode} does not indicate success.");
+            _logRetrieveCacheEntryUsingBaseKey = LoggerMessage.Define<string>(
+                logLevel: LogLevel.Debug,
+                eventId: 18,
+                formatString: "Retrieving cache entry for base key {BaseKey}.");
+            _logRetrieveCacheEntryUsingVaryByKey = LoggerMessage.Define<string>(
+                logLevel: LogLevel.Debug,
+                eventId: 19,
+                formatString: "Retrieving cache entry for varyby key {VaryByKey}.");
+            _logIfNoneMatchStarSend304 = LoggerMessage.Define(
+                logLevel: LogLevel.Debug,
+                eventId: 20,
+                formatString: $"The content requested is not modified. The {HeaderNames.IfNoneMatch} header of the request contains a value of *.");
+            _logIfNoneMatchEtagSend304 = LoggerMessage.Define<EntityTagHeaderValue>(
+                logLevel: LogLevel.Debug,
+                eventId: 21,
+                formatString: $"The content requested is not modified. The ETag {{ETag}} in the {HeaderNames.IfNoneMatch} header matched the ETag of a cached entry.");
+            _logIfUnmodifiedSinceSend304 = LoggerMessage.Define<DateTimeOffset, DateTimeOffset>(
+                logLevel: LogLevel.Debug,
+                eventId: 22,
+                formatString: $"The content requested is not modified. The last modified date of {{LastModified}} is before the date {{IfUnmodifiedSince}} specified in the {HeaderNames.IfUnmodifiedSince} header.");
+            _log304NotModifiedServed = LoggerMessage.Define(
+                logLevel: LogLevel.Information,
+                eventId: 23,
+                formatString: $"The content requested is not modified. A {StatusCodes.Status304NotModified} {ReasonPhrases.GetReasonPhrase(StatusCodes.Status304NotModified)} response will be served.");
+            _logCachedResponseServed = LoggerMessage.Define(
+                logLevel: LogLevel.Information,
+                eventId: 24,
+                formatString: "A valid cached response was retrieved for this request. The cached response will be served.");
+            _log504GatewayTimeoutServed = LoggerMessage.Define(
+                logLevel: LogLevel.Information,
+                eventId: 25,
+                formatString: $"No valid response was retrieved for this request but a only-if-cached cache directive was specified. A {StatusCodes.Status504GatewayTimeout} {ReasonPhrases.GetReasonPhrase(StatusCodes.Status504GatewayTimeout)} response will be served.");
```

Comment:
"No cached response available for this request and the only-if-cached cache directive was specified."?